editor/blocks/list/index.js

```diff
@@ -0,0 +1,43 @@
+const Editable = wp.blocks.Editable;
+const { html, prop } = wp.blocks.query;
+
+function List( { nodeName, children } ) {
+	// nodeName.toLowerCase() is used to map DOM nodeName values to proper tag.
+	return wp.element.createElement( nodeName.toLowerCase(), null, children );
+}
+
+wp.blocks.registerBlock( 'core/list', {
+	title: 'List',
+	icon: 'editor-ul',
+	category: 'common',
+
+	attributes: {
+		listType: prop( 'ol,ul', 'nodeName' ),
+		items: wp.blocks.query.query(
+			'li',
+			{
+				value: html()
+			}
+		)
+	},
+
+	edit( { attributes, onChange } ) {
+		const { listType = 'ol', items = [] } = attributes;
+		const value = items.map( i => {
+			return `<li>${ i.value }</li>`;
+		} ).join( '' );
+
+		return (
+			<Editable
+				nodeName={ listType }
+				value={ value }
+				onChange={ onChange } />
+		);
+	},
+
+	save( { attributes } ) {
+		const { listType = 'ol', items = [] } = attributes;
+		const children = items.map( ( i, index ) => <li key={ index }>{ i.value }</li> );
```

Comment:
We'll probably need to `dangerouslySetInnerHTML` here for HTML in `value` to not be escaped. I'd like to explore options here to make this less... scary, and more obvious / transparent.

```js
const children = items.map( ( item, index ) => (
	<li dangerouslySetInnerHTML={ { __html: item.value } } />
) );
```

I'd be curious to know whether `key` matters at this line since we're not doing any sort of reconciliation, we're simply calling `renderToString` at this point.